common/web/gesture-recognizer/src/engine/eventSequentializationQueue.ts

```diff
@@ -0,0 +1,51 @@
+import { timedPromise } from "@keymanapp/web-utils";
+
+export class EventSequentializationQueue {
+  private queue: (() => Promise<void> | void)[];
+  private defermentPromise: Promise<void>;
+
+  constructor() {
+    this.queue = [];
+  }
+
+  private setDeferment(promise: Promise<any>) {
+    this.defermentPromise = promise;
+    promise.then(() => {
+      this.defermentPromise = null;
+      this.triggerEvent();
+    });
+  }
```

Comment:
You use async for triggerEvent, why not here also?

```suggestion
  private async setDeferment(promise: Promise<any>) {
    this.defermentPromise = promise;
    await promise;
    this.defermentPromise = null;
    this.triggerEvent();
  }
```

(see other comments for name of 'defermentPromise', 'setDeferment', etc)